wandb/util.py

```diff
@@ -472,7 +472,11 @@ def is_jax_tensor_typename(typename: str) -> bool:
 def get_jax_tensor(obj: Any) -> Optional[Any]:
     import jax  # type: ignore
 
-    return jax.device_get(obj)
+    jax_tensor = jax.device_get(obj)
+    # convert bfloat16 to float32
+    if jax_tensor.dtype == jax.dtypes.bfloat16:
+        return jax_tensor.astype("float32")
```

Comment:
if we actually want to do the conversion for the user I think we should do it in the `json_friendly` function, because this function just gets the `jax` tensor so we add an unexpected conversion (if someone wants to use it in another capacity). Also do we want to have warning about the conversion?
I think it is ok to do it silently if we do it in the `json_friendly` since this type is not supported for conversion.

Reply:
In `json_friendly` you can just check that the `numpy` tensor doesn't have `bfloat16` type and convert it... until `numpy` decides to support this type 😆

Reply:
Example where your current fix would break:

```
m1 = tf.constant(1., dtype='bfloat16')
run.log(dict(m1=m1))
```

I think for PyTorch it will fail to do the conversion to numpy with `dtype`- `bfloat16`

Reply:
many thanks for this suggestion -- implemented what you suggested and now it all works :)